apps/cowswap-frontend/src/modules/trade/hooks/useNavigateOnCurrencySelection.ts

```diff
@@ -80,12 +81,22 @@ export function useNavigateOnCurrencySelection(): CurrencySelectionCallback {
        */
       const shouldResetBuyToken = isInputField && targetChainMismatch
       const shouldSetTargetChain = !isInputField && targetChainMismatch
+      const shouldResetBuyOrder = !isInputField && orderKind === OrderKind.BUY
```

Comment:
Shouldn't this be done only if this is a bridge swap?

Reply:
Nice catch! Thanks!
https://github.com/cowprotocol/cowswap/pull/5588/commits/e05acf33b066a303afa3d23159a3b6b937fe8e0c